akka-persistence/src/main/scala/akka/persistence/Persistent.scala

```diff
@@ -36,20 +36,24 @@ object AtomicWrite {
 
 final case class AtomicWrite(payload: immutable.Seq[PersistentRepr]) extends PersistentEnvelope with Message {
   require(payload.nonEmpty, "payload of AtomicWrite must not be empty!")
+  private var optHighestSequenceNr: Option[Long] = None
```

Comment:
perhaps just `Long = 0`, since the payload will always be non empty it will always have at least one value set by the line below (in the forall)